server/methods/getSetupWizardParameters.js

```diff
@@ -5,11 +5,12 @@ import { Settings } from '../../app/models';
 Meteor.methods({
 	getSetupWizardParameters() {
 		const settings = Settings.findSetupWizardSettings().fetch();
-		const allowStandaloneServer = process.env.DEPLOY_PLATFORM !== 'rocket-cloud';
+		const [registerServerSetting] = Settings.findById('Register_Server').fetch();
+		const canSkipRegistration = registerServerSetting.value || process.env.DEPLOY_PLATFORM === 'rocket-cloud';
```

Comment:
I don't think this will cover what you're trying to do.  

Register_Server is a boolean that controls if a user "wants" the server to be registered or allowed to communicate with Cloud  It doesn't reflect in any way the actual registration status of the workspace.

The reliable indicator that a workspace is actually registered would be: 
https://github.com/RocketChat/Rocket.Chat/blob/7632f12cfcc7ed8ee8f843587fdff63b101cc765/app/cloud/server/functions/retrieveRegistrationStatus.js#L7

```
settings.get('Cloud_Workspace_Client_Id')
```

You'll notice this is the `settings` and not the `Settings` also.  Need to use the raw in this case otherwise we will be back here fixing this having bugs in a few days.

The cached Settings isn't the most reliable in this case.  Because the registration could have happened on another instance and some how not propagated to the cache via oplog/change stream.  

Before we made the change to the raw settings it would randomly return false when it was actually true.  Causing a lot of issues as you can imagine